src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/FeatureFlag/FeatureFlagsFormDataProvider.php

```diff
@@ -131,4 +135,28 @@ protected function getOneFeatureFlagByName(string $featureFlagName): ?FeatureFla
     {
         return $this->doctrineEntityManager->getRepository(FeatureFlag::class)->findOneBy(['name' => $featureFlagName]);
     }
+
+    // conditions the display of the AuthorizationServerMultistore feature fag only if AuthorizationServer is activated, regardless of its stability
+    private function checkAuthorizationServerMultistore(array $featureFlagsData): array
+    {
+        $authorizationServerMultistoreKey = FeatureFlagSettings::FEATURE_FLAG_AUTHORIZATION_SERVER_MULTISTORE;
+        $authorizationServerKey = FeatureFlagSettings::FEATURE_FLAG_AUTHORIZATION_SERVER;
+        $isMultistoreActive = $this->multiStoreFeature->isActive();
+
+        if (array_key_exists($authorizationServerMultistoreKey, $featureFlagsData)) {
+            if (!$isMultistoreActive) {
+                unset($featureFlagsData[$authorizationServerMultistoreKey]);
+            } elseif (array_key_exists($authorizationServerKey, $featureFlagsData)
+                && !$featureFlagsData[$authorizationServerKey]['enabled']) {
+                unset($featureFlagsData[$authorizationServerMultistoreKey]);
+            } else {
```

Comment:
Maybe this block is overkill, since you check the value via `isEnabled` right under, it could be enough as check to simplify the code here no?
```suggestion
            } else {
```

Reply:
there is an interest. If there is a different stability between the 2 feature flags, you must use a request, otherwise they are in the same list. This saves us a request